### PR TITLE
Adds iLabs Challenger boards.

### PR DESCRIPTION
**Pull request by PontusO**
This PR adds the following boards:
  - Challenger+ RP2350 WiFi/BLE5
  - Challenger+ RP2350 BConnect

### CODE REVIEW

**Review by tannewt — CHANGES_REQUESTED** [review]
One pre-commit issue. Good otherwise!

File: ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+CHIP_FAMILY = rp2
+
+EXTERNAL_FLASH_DEVICES = "W25Q64JVxQ"
+
+CIRCUITPY__EVE = 1
+CIRCUITPY_ALARM = 0
+
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_HID
+FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_Register
+
```
[the remark this answers is not in the document]
Comment (PontusO) [reply ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk:17]: Not sure what I need to do here, github says I can't commit the suggested change =)
gitcommit

[the remark this answers is not in the document]
Comment (tannewt) [reply ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk:17]: Weird! The newline at the end needs to be removed.

[the remark this answers is not in the document]
Comment (PontusO) [reply ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk:17]: That is so odd. On the first commit it complained I was missing a newline and now it wants to get rid of it 😏 I'll revisit my fork tomorrow and try to figure out what i'm doing wrong here.

[the remark this answers is not in the document]
Comment (tannewt) [reply ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk:17]: I'd suggest installing pre-commit locally. It'll fix it for you.

**Comment by dhalbert** [comment]
If you are editing files through the GitHub web interface, it is sometimes hard to get the last lines correct. Pushing from a local copy will make sure that what you think is in the file is really there.

File: ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (PontusO) [reply ports/raspberrypi/boards/challenger_rp2350_wifi6_ble5/mpconfigboard.mk]: Yeah, I dont know what to do now. I installed pre-commit and it fixed the files. ran git add ., git commit and pushed it. Still no dice .

**Comment by dhalbert** [comment]
Don't worry about the doc build failure -- we know about that and it is unrelated.

**Comment by jepler** [comment]
The good news is, your last push did resolve the pre-commit problem during CI.

The failure was in building the documentation, which is not due to your PR. It can be ignored. (should be fixed by #9580)

The other problem that needs to be fixed is that your latest change also changes the Pico-PIO-USB submodule. Using `git add .` or `git commit -a` (rather than individually choosing each file to commit in a program like git gui) is very prone to including more than you intended to change, especially in the case of submodules, due to other design choices in git (just doing a `git switch` or `git checkout` does not update submodules, meaning that they become marked as changed)

I made an attempt to fix this but I was not permitted by github to add my change to this PR:
```
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
remote: This repository moved. Please use the new location:
remote:   git@github.com:PontusO/circuitpython.git
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: error: Changes must be made through a pull request.
To github.com:pontuso/circuitpython
 ! [remote rejected]       pr9562 -> main (protected branch hook declined)
```
so, you will have to resolve this. The steps I followed were roughly:
 * open a shell on ports/raspberrypi/lib/Pico-PIO-USB
 * `git fetch origin`
 * `git checkout fe9133f`
 * `cd ..`
 * `git commit -m"Pico-PIO-USB: undo unintended change to submodule" -- Pico-PIO-USB`

once you've done these steps and pushed, you should be able to look at the list of files changed on github and see that Pico-PIO-USB is not listed.

I can't recommend strongly enough to use a tool (whether command-line or GUI) that individually selects which files to commit, to avoid irritations like this.

**Comment by bablokb** [comment]
One other thing maybe worth changing for the challenger_rp2350_wifi6_ble: `board.ESP_RXD` is actually the TX-pin (from the perspective of the MCU). Same with `board.ESP_TXD`, which is the RX-pin. The very similar challenger_rp2040_wifi_ble has this right (but using the pin-names `board.ESP_RX`, `board.ESP_TX`).

I would also keep the pin-names in sync. I assume there will be users upgrading from the rp2040_wifi_ble and they will certainly be more happy if they don't need to change their programs.

**Review by tannewt — APPROVED** [review]
Thank you! Please also make a PR to https://github.com/raspberrypi/usb-pid to publicly document the PIDs.

**Pull request merged**